src/map/entities/charentity.cpp

```diff
@@ -2134,7 +2134,16 @@ void CCharEntity::OnRaise()
             // Player died within a battlefield, reward the battlefield level equivalent EXP
             if (StatusEffectContainer->HasStatusEffect(EFFECT_BATTLEFIELD))
             {
-                expLost = m_LevelRestriction >= 1 && m_LevelRestriction <= 67 ? (charutils::GetExpNEXTLevel(m_LevelRestriction) * 8) / 100 : 2400;
+                if (PBattlefield != nullptr && (PBattlefield->GetRuleMask() & RULES_LOSE_EXP) != RULES_LOSE_EXP)
+                {
+                    expLost = 0;
```

Comment:
Players won't lose EXP, sure. But this wouldn't prevent them from gaining EXP when they're raised. Therefore creating an exploitable infinite loop of kill -> raise -> gain 2K exp repeat.

Reply:
expLost has nothing to do with losing exp. It is a variable in onRaise() that calculates how much exp to return to the player based off how much they would have lost.

Reply:
Great! I appreciate the DM.